modules/infoserv/main.c

```diff
@@ -287,35 +258,27 @@ is_cmd_post(struct sourceinfo *si, int parc, char *parv[])
 		return;
 	}
 
-	if (imp == 0)
-	{
-		o = smalloc(sizeof *o);
-		o->nick = strshare_ref(entity(si->smu)->name);
-		o->info_ts = CURRTIME;
-		o->story = sstrdup(story);
-		o->subject = sstrdup(subject);
-
-		n = mowgli_node_create();
-		mowgli_node_add(o, n, &operlogon_info);
-	}
+	struct logoninfo *l = smalloc(sizeof *l);
+	l->nick = strshare_ref(entity(si->smu)->name);
+	l->info_ts = CURRTIME;
+	l->story = sstrdup(story);
+	l->subject = sstrdup(subject);
 
-	if (imp > 0)
-	{
-		l = smalloc(sizeof *l);
-		l->nick = strshare_ref(entity(si->smu)->name);
-		l->info_ts = CURRTIME;
-		l->story = sstrdup(story);
-		l->subject = sstrdup(subject);
-
-		n = mowgli_node_create();
-		mowgli_node_add(l, n, &logon_info);
-	}
+	mowgli_list_t *list = (imp == 0) ? &operlogon_info : &logon_info;
+	mowgli_node_t *n = mowgli_node_create();
+	mowgli_node_add(l, n, list);
 
 	command_success_nodata(si, _("Added entry to logon info"));
+	if (logoninfo_count != 0 && !logoninfo_reverse && list->count > logoninfo_count)
+		command_success_nodata(si, ngettext("Warning: There is more than %u entry, so your message will not be displayed by default.",
```

Comment:
These strings need to be wrapped in `N_()` so that the `update-pot` target in `po/` can find them.

Reply:
`xgettext` handles `ngettext()` calls by default, including proper handling of the pluralization stuff. That said, it does look like our `update-potfiles.sh` script doesn't check for calls to it. In practice, most files using `ngettext()` will also use `_()` somewhere in the code, but we probably should add a check for that.

Reply:
My bad; I misunderstood the documentation for the `--keyword` option in `xgettext(1)`. If you specify it with no value, *then* it inhibits the default keywords like `ngettext`. However, we only specify *additional* keywords, so it will (and does) find these strings.